### ISSUE TITLE
Compiling for Windows

### ISSUE THREAD

**Issue by ShazAu** [WyvernTKC/cpuminer-gr-avx2#143]
Hi,
I'm trying to compile the miner for Windows and have run into trouble.  I've been following the instructions found in the INSTALL_WINDOWS.  I'm a windows guy so I found this hard to follow.  I'll give a brief summary of the current issue and then a full set of steps that I followed to get this far.  Apologies for the wall of text!

Under step 4 
copying files to the releases folder I have 3 files that fail to copy
This copy fails
_cp $GCC_MINGW_LIB/libstdc++-6.dll release/_
cp: cannot stat '/usr/lib/gcc/x86_64-w64-mingw32/10-win32/libstdc++-6.dll': No such file or directory

This copy also fails 
_cp $GCC_MINGW_LIB/libgcc_s_seh-1.dll release/_
cp: cannot stat '/usr/lib/gcc/x86_64-w64-mingw32/10-win32/libgcc_s_seh-1.dll': No such file or directory

This copy also failed
_cp $LOCAL_LIB/openssl/libcrypto-1_1-x64.dll release/_
The above file doesn't exist even though the make of openssl did not report errors, copied the following file instead, thought it might be a versioning thing:
_cp $LOCAL_LIB/openssl/libcrypto-3-x64.dll release/_

Trying to compile the application as is causes fatal errors, no such file or directory for: 
curl/curl.h
openssl/sha.h

Complete build steps so far:

I'm running the latest version of Linux Mint xfce (20.2 Uma) 64bit
Installed the pre reqs:
_sudo apt-get install build-essential automake libssl-dev libcurl4-openssl-dev libjansson-dev libgmp-dev zlib1g-dev libnuma-dev git_
Cloned the source from Git:
_git clone https://github.com/WyvernTKC/cpuminer-gr-avx2_
Installed mingw-w64:
_sudo apt-get install mingw-w64 libz-mingw-w64-dev_
Created the $HOME/usr/lib folder, ($HOME/shaz/lib in my case:
_mkdir lib_
Changed directory to the lib folder
_cd lib_
Cloned openssl from Git:
_git clone https://github.com/openssl/openssl/_
Cloned curl from Git:
_git clone https://github.com/curl/curl_
Downloaded gmp and extracted to $HOME/shaz/lib/gmp/gmp-6.2.1

Configured mingw64 options
_./Configure mingw64 shared --cross-compile-prefix=x86_64-w64-mingw32-_
Compiled mingw64 (using 3 cores)
_make -j 3_

Curl, changed directory to curl folder
_cd
cd lib/curl/_
Install additional pre-requisites
_sudo apt-get install autoconf automake libtool perl
Configure Curl
autoreconf -fi
./configure --with-winssl --with-winidn --with-schannel --host=x86_64-w64-mingw32
Compile curl (using 3 cores)
make -j 3_

gmp, changed directory to gmp folder
cd
cd lib/gmp/gmp-6.2.1
./configure --host=x86_64-w64-mingw32
make -j 3

export LOCAL_LIB="$HOME/usr/lib"
export CONFIGURE_ARGS="--with-curl=$LOCAL_LIB/curl --with-crypto=$LOCAL_LIB/openssl --host=x86_64-w64-mingw32"
export MINGW_LIB="/usr/x86_64-w64-mingw32/lib"
export GCC_MINGW_LIB="/usr/lib/gcc/x86_64-w64-mingw32/10-win32"

Make the release folder
cd
mkdir release

Copy files
_cp $MINGW_LIB/zlib1.dll release/
cp $MINGW_LIB/libwinpthread-1.dll release/_

This copy fails
_cp $GCC_MINGW_LIB/libstdc++-6.dll release/_
cp: cannot stat '/usr/lib/gcc/x86_64-w64-mingw32/10-win32/libstdc++-6.dll': No such file or directory
This copy also fails 
_cp $GCC_MINGW_LIB/libgcc_s_seh-1.dll release/_
cp: cannot stat '/usr/lib/gcc/x86_64-w64-mingw32/10-win32/libgcc_s_seh-1.dll': No such file or directory

This copy also failed
_cp $LOCAL_LIB/openssl/libcrypto-1_1-x64.dll release/_
file doesn't exist even thought make of openssl did not report errors, copied the following file instead, thought it might be a versioning thing
_cp $LOCAL_LIB/openssl/libcrypto-3-x64.dll release/_

$LOCAL_LIB/curl/lib/.libs/libcurl-4.dll release/

Step 6. after here I don't have exact steps I followed...
Changed folder to cpuminer-gr-avx2
_cd
cd cpuminer-gr-avx2_
export LDFLAGS="-L$LOCAL_LIB/curl/lib/.libs -L$LOCAL_LIB/gmp/.libs -L$LOCAL_LIB/openssl"
rm ./gmp.h 2>/dev/null
ln -s $LOCAL_LIB/gmp/gmp.h ./gmp.h

This command fails:
**make distclean**
**make: *** No rule to make target 'distclean'.  Stop.**
(I think this is just because I haven't run make before?)

rm -f config.status
./autogen.sh

export DCFLAGS="-fno-common -D_WIN32_WINNT=0x0601"
CFLAGS="-O3 -march=native -mtune=native ${DCFLAGS}" \
CXXFLAGS="$CFLAGS -std=c++20 -Wno-ignored-attributes" \
./configure ${CONFIGURE_ARGS}

Compile, using 3 threads
make -j 3

compilation error:
In file included from sysinfos.c:17,
                 from util.c:15:
miner.h:42:10: fatal error: curl/curl.h: No such file or directory
   42 | #include <curl/curl.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:1255: cpuminer-util.o] Error 1
make[2]: *** Waiting for unfinished jobs....
api.c:22:10: fatal error: openssl/sha.h: No such file or directory
   22 | #include <openssl/sha.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
In file included from ./algo-gate-api.h:4,
                 from algo/sha/sha256d.h:1,
                 from cpu-miner.c:27:
./miner.h:42:10: fatal error: curl/curl.h: No such file or directory
   42 | #include <curl/curl.h>
      |          ^~~~~~~~~~~~~
compilation terminated.

**Comment by adekkpl** [comment]
I never tried to compile it but think that something is wrong with curl and openssl missing libraries, have You tried thease solutions?
https://stackoverflow.com/questions/11471690/curl-h-no-such-file-or-directory/11471743   for curl
https://unix.stackexchange.com/questions/87458/make-fatal-error-openssl-sha-h-no-such-file-or-directory  for ssl

**Comment by qwewer0** [comment]
@ShazAu Hope this helps: [INSTALL_WINDOWS.md](https://github.com/qwewer0/cpuminer-gr-avx2_Fair-Donation/blob/main/INSTALL_WINDOWS.md)

**Comment by ShazAu** [comment]
Thanks for the replies guys.  I'm currently playing with the xmrig miner at the moment so I'm going to close this issue for now.
